server/src/main/java/org/elasticsearch/snapshots/RestoreService.java

```diff
@@ -655,16 +655,14 @@ private void ensureNotPartial(String index) {
             }
         }
 
-        private void validateExistingIndex(IndexMetadata currentIndexMetadata, IndexMetadata snapshotIndexMetadata, String renamedIndex) {
+        private void validateExistingIndex(RelationName targetName, IndexMetadata currentIndexMetadata, IndexMetadata snapshotIndexMetadata, String renamedIndex) {
```

Comment:
We could also remove the `renamedIndex` argument and adjust the later exception to just use the `targetName` (or we'd endode it again, but not sure if the encoded partition ident is already useful inside the error)